File: iOS-ProjectStructureDemo/ExampleProjectDemo/CodeFeed/EntityCode/Home/C/BaseHomeViewController.m
//
//  BaseHomeViewController.m
//  ExampleProjectDemo
//
//  Created by BlanBok on 2018/1/15.
//  Copyright © 2018年 juyuanGroup. All rights reserved.
//

#import "BaseHomeViewController.h"
#import "BaseHomeDataModelIndex.h"
#import "BaseHomeViewIndex.h"
#import "BaseHomeViewModelIndex.h"

@interface BaseHomeViewController ()

@property (nonnull,nonatomic,strong) BaseHomeDataModelIndex *index_dataModel;

@property (nonnull,nonatomic,strong) BaseHomeViewIndex *index_view;

@property (nonnull,nonatomic,strong) BaseHomeViewModelIndex *index_viewModel;


@end

@implementation BaseHomeViewController


- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.view.backgroundColor = CUSTOM_COLOR_HEX(0xf4f4f4);
    [self init_commit];
}

- (void)init_commit
{
    [self setIndex_dataModel];
    [self setIndex_view];
    [self setIndex_viewModel];
}

#pragma mark-init
- (void)setIndex_dataModel
{
    
}

- (void)setIndex_view
{
    self.index_view.container = self.view;
}

- (void)setIndex_viewModel
{
    
}

#pragma mark-lazyload
- (BaseHomeDataModelIndex *)index_dataModel
{
    if (_index_dataModel == nil) {
        _index_dataModel = [BaseHomeDataModelIndex new];
    }
    return _index_dataModel;
}

- (BaseHomeViewIndex *)index_view
{
    if (_index_view == nil) {
        _index_view = [BaseHomeViewIndex new];
    }
    return _index_view;
}

- (BaseHomeViewModelIndex *)index_viewModel
{
    if (_index_viewModel == nil) {
        _index_viewModel = [BaseHomeViewModelIndex new];
    }
    return _index_viewModel;
}


@end

File: DemoPreoject/DemoPreoject/CodeFeed/EntityCode/ModuleHome/home/C/BaseHomeViewController.swift
//
//  BaseHomeViewController.swift
//  DemoPreoject
//
//  Created by 小虎 on 2019/1/7.
//  Copyright © 2019 OComme. All rights reserved.
//
//BaseHome生命周期控制及业务中转站

import UIKit

class BaseHomeViewController: UIViewController {

    private let index_Model:BaseHomeModelIndex = BaseHomeModelIndex.init()
    
    private let index_view:BaseHomeViewIndex = BaseHomeViewIndex.init()

    private let index_viewModel:BaseHomeViewModelIndex = BaseHomeViewModelIndex.init()

    
    // MARK: - life cycle
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        setindex_Model()
        setIndex_view()
        setIndex_viewModel()
    }
    
    // MARK: - init
    /// 数据模型初始配置
    private func setindex_Model() {
        
    }
    
    /// 显示层初始配置
    private func setIndex_view() {
        self.index_view.setContainer(self.view)

    }
    
    /// 配置逻辑及业务逻辑初始配置
    private func setIndex_viewModel() {
        
    }
}
